packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -623,7 +623,11 @@ class MarkdownBuilder implements md.NodeVisitor {
         final RichText previous = mergedTexts.removeLast() as RichText;
         final TextSpan previousTextSpan = previous.text as TextSpan;
         final List<TextSpan> children = previousTextSpan.children != null
-            ? List<TextSpan>.from(previousTextSpan.children!)
+            ? previousTextSpan.children!
+                .map((InlineSpan span) => span is! TextSpan
+                    ? TextSpan(children: <InlineSpan>[span])
+                    : span)
+                .toList()
             : <TextSpan>[previousTextSpan];
         children.add(child.text as TextSpan);
         final TextSpan? mergedSpan = _mergeSimilarTextSpans(children);
```

Comment:
@edhom Are you sure is the right place to fix this? I am wondering _why_ did the code previously assume that these are all `TextSpan`s?

I am thinking we should maybe change `_maybeSimilarTextSpans` to `_mergeSimilarInlineSpans` and fix the types that way. Then the tests would also be easier because we know we just need to place different `InlineSpan`s as children.

Reply:
I can't answer why we wound up here, but I like the suggested action here.